### PR TITLE
Improve documentation of kernel

### PR DESCRIPTION
**Pull request by privat**
Improve the documentation of Object and Sys.
Also experiments with the new nitdoc features #1447

Some issues I discovered while writing and using the doc while thinking about nitdoc.

* micro-picking classes and methods to include in the README is not fun
* headings (`###`) in the markdown might not have the right level in the generated HTML
* having reified property groups (#1413) should help to organize thing and have automatic feedback on it trough the doc
* maybe the `[[list:]]` directive should no list entities not already included by `[[doc:]]` (or a new `[[list_other:]]` directive?)
* having a way to generate a full linear and strucuted documentation in a single page could simplify the job of the writer when checking the doc.

### CODE REVIEW

**Comment by Morriar** [comment]
Can you elaborate on the last issue?

**Comment by privat** [comment]
@Morriar : just having a big HTML document with sections everything inside: groups, modules, classes, properties, etc.

**Comment by Morriar** [comment]
One idea reading the documentation of methods like `==`, `!=`.

Since we always use a concrete name in the compiler and model, why not reify this name in the mproperty so it can be displayed in the doc:

[==](#) (equals) — Have `self` and `other` the same value?

**Comment by Morriar** [comment]
Yeah I see waht you mean with: `[[doc:main]]`. It should be fixed once the concept generalized to all MDoc entities.

**Comment by Morriar** [comment]
It's cool, +1

**Comment by Morriar** [comment]
I like the idea of the `other_list` directive. It can also help for micro picking since you can see what you have not documented yet.

**Comment by privat** [comment]
need reviews. demo here if some prefer: http://info.uqam.ca/~privat/demodoc/

File: lib/standard/kernel.nit [hunk 1]
```diff
@@ -27,9 +30,13 @@ in "C" `{
 ###############################################################################
 
 # The root of the class hierarchy.
-# Each class implicitly specialize Object.
 #
-# Currently, Object is also used to collect all top-level methods.
+# Each other class implicitly specializes Object,
+# therefore the services of Object are inherited by every other classes and are usable
```
Comment (R4PaSs) [inline lib/standard/kernel.nit]: :s/classes/class

File: lib/standard/kernel.nit [hunk 2]
```diff
@@ -53,33 +60,124 @@ interface Object
...
+	# It is also used by the `inspect` method to loosely identify objects
+	# and helps debugging.
+	#
+	# ~~~
+	# var a = "Hello"
+	# var b = a
+	# assert a.object_id == b.object_id
+	# ~~~
+	#
+	# The specific details of the internal hash code it let to the specific
```
Comment (R4PaSs) [inline lib/standard/kernel.nit]: :s/it let to/is left to/ ?

File: lib/standard/kernel.nit [hunk 3]
```diff
@@ -53,33 +60,124 @@ interface Object
...
+	# engine. The rules are the following:
+	#
+	# * The `object_id` MUST be invariant for the whole life of the object.
+	# * Two living instances of the same classes SHOULD NOT share the same `object_id`.
+	# * Two instances of different classes MIGHT share the same `object_id`.
+	# * The `object_id` of a garbage-collected instance MIGHT be reused by new instances.
+	# * The `object_id` of an object MIGHT be non constant across different executions.
+	#
+	# For instance, the `nitc` compiler choose to use the address of the object in memory
```
Comment (R4PaSs) [inline lib/standard/kernel.nit]: :s/choose/chooses

File: lib/standard/kernel.nit [hunk 4]
```diff
@@ -88,26 +186,117 @@ interface Object
...
+	# only and can be removed without any notice.
+	#
+	# TODO: rename to avoid blocking a good identifier like `output`.
 	fun output_class_name is intern
 
 	# The hash code of the object.
-	# Assuming that a == b -> a.hash == b.hash
 	#
-	# Without redefinition, it is based on the `object_id` of the instance.
+	# The hash code is used in many data-structures and algorithms to identify objects that might be equals.
```
Comment (R4PaSs) [inline lib/standard/kernel.nit]: :s/equals/equal

File: lib/standard/kernel.nit [hunk 5]
```diff
@@ -88,26 +186,117 @@ interface Object
...
+	# `hash` (like `==`) might not be constant on some objects over time because of their evolution.
+	#
+	# ~~~
+	# var a = [1]
+	# var b = [1]
+	# var c = [1,2]
+	# assert a.hash == b.hash
+	# a.add 2
+	# assert a.hash == c.hash
+	# # There is a very hight probability that `b.hash != c.hash`
```
Comment (R4PaSs) [inline lib/standard/kernel.nit]: :s/hight/high

**Comment by R4PaSs** [comment]
Aside from the few typos, :shipit: !

**Comment by R4PaSs** [comment]
Side-note: why are the links in the Program Execution section still in their wikilinks from ?

**Comment by privat** [comment]
reroll: applied type fixes and updated sav files for tests

**Pull request merged**